File: myfirstproject-user-web/src/main/webapp/js/service/userService.js
//服务层
app.service('userService',function($http){
	    	
    this.register=function (user,encode) {
        return $http.post('user/add.do?encode='+encode,user);
    }

    this.sendCode=function (phone) {
        return $http.get('user/sendCode.do?phone='+phone);
    }
});

File: myfirstproject-user-interface/src/main/java/com/pinyougou/user/service/UserService.java
package com.pinyougou.user.service;

import com.pinyougou.pojo.TbUser;
import entity.PageResult;

import java.util.List;

public interface UserService {

    public List<TbUser> findAll();

    public PageResult findPage(int pageNum, int pageSize);

    public void add(TbUser user);

    public void update(TbUser user);

    public TbUser findOne(Long id);

    public void delete(Long[] ids);

    public PageResult findPage(TbUser user, int pageNum, int pageSize);

    /**
     * 发送短信验证码
     * @param phone
     */
    void sendCode(String phone);

    /**
     * 比较验证码
     * @param phone
     * @param code
     * @return
     */
    public boolean checkCode(String phone,String code);
}
